modules/storage/filesystem/device.go

```diff
@@ -62,45 +62,49 @@ func (d *Device) Used() bool {
 // Found devices are cached, and the cache is only repopulated after the `Scan`
 // method is called.
 type lsblkDeviceManager struct {
-	devices []Device
+	devices DeviceCache
 }
 
 // DefaultDeviceManager returns a default device manager implementation
-func DefaultDeviceManager() DeviceManager {
-	return &lsblkDeviceManager{
-		devices: []Device{},
+func DefaultDeviceManager(ctx context.Context) (DeviceManager, error) {
+	m := &lsblkDeviceManager{
+		devices: DeviceCache{},
 	}
+
+	err := m.scan(ctx)
```

Comment:
While I like the idea of a `one time` scan of the devices on module load. This can cause an issue when if the machine disks has been replaced (hot plugged) while the machine is running. I have no idea if this is a case that we have to care about at this point.

Reply:
We will need to supports this at some point. This was planned in the original interface of the storage daemon.
I don't know yet exactly how we're going to implement this. But yes, we want to be able to hot swap disk in a node.
But we can probably planned that for later release

Reply:
I would assume that hot swapping disks generates some kernel message, so we can listen for those and act accordingly in the future